docs/structured-streaming-programming-guide.md

```diff
@@ -1126,6 +1126,12 @@ aggDF \
 </div>
 </div>
 
+## Advanced Sources
```

Comment:
Discussed offline. Best to move it to the same location where the File and socket sources are discussed. Insert a subheading called Input Sources to mirror the subheading Output Sinks.